### DISCUSSION TITLE
Help with custom Sleep command

### DISCUSSION THREAD

**Discussion by fullklax** [Open-Shell/Open-Shell-Menu#1956] [Q&A] [answered]
I have VBScripts playing a sound that fire up when I: Log on, Lock, and Unlock my PC (via task scheduler). I'm on Win10.

However, when I put my PC to sleep, it borks the 'lock' script, because the sound doesn't have time to play before the PC is suspended, so I get an error after I wake it up and unlock it. And I've found no way to delay the sleep command, other than firing it myself manually, when I want it to (at the press of a 'Sleep' button, but only after executing some VB code).

So I figured this could work for me: I'd like to know the command used by OpenShell to put the PC to sleep (shown below), so I can use it in my script.

![image](https://github.com/user-attachments/assets/51aebcb0-2140-43e3-999b-90800ce972e6)

Also I believe this may help you identify exactly what kind of sleep or suspend I want to fire off:

![image](https://github.com/user-attachments/assets/c7c9f424-f6fb-40f6-b8af-5d5a1a5989c9)

Thank you for looking into this, and for maintaining such a great, and IMO indispensable utility <3

**Comment by ge0rdi** [comment, marked as answer]
> I'd like to know the command used by OpenShell to put the PC to sleep (shown below), so I can use it in my script.

[SetSuspendState](https://learn.microsoft.com/en-us/windows/win32/api/powrprof/nf-powrprof-setsuspendstate) API is used on standard machines:
`SetSuspendState(FALSE, FALSE, FALSE);`

On machines with connected standby there is [NtPowerInformation](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-ntpowerinformation) used:
`NtPowerInformation(ScreenOff, NULL, 0, NULL, 0);`

Though you can also invoke `Open-Shell`'s `sleep` command directly via:
`"C:\Program Files\Open-Shell\StartMenu.exe" -cmd sleep`

**Reply by fullklax** [reply]
Awesome, that works, thank you! I went the more round-about but simple way of using your implementation of it as per your last suggestion, and yep, works as intended.

Now I have another issue where I still want my task sheduler script to fire "at workstation lock" (when I hit Windows+L), but Sleep triggers a lock too, so now it fires twice, ugh... Guess I'll have to figure a way to fire it elsewhere than from task scheduler. Thanks again!
